manifests/server.pp

```diff
@@ -158,18 +159,43 @@
           }
         }
       }
+    }
 
-      # Wrap the replset class
-      class { 'mongodb::replset':
-        sets => $replset_config_REAL
-      }
-      Anchor['mongodb::server::end'] -> Class['mongodb::replset']
+    # Wrap the replset class
+    class { 'mongodb::replset':
+      sets => $replset_config_REAL
```

Comment:
If `$replset` is not defined, this would be undef, yeah? Under what case should this be passed an undefined `sets`?